### DISCUSSION TITLE
Using a workspace file in Codespaces?

### DISCUSSION THREAD

**Discussion by brwilkinson** [community/community#6843] [Codespaces] [answered]
Is there any reason to load a workspace file in a Codespace?

- Can I just load the workspace file on Codespace launch via startup command?
    - Thinking mainly for settings, not to duplicate these between Codespace users and local users?
    - Also for possible optional extension recommendations?

**Comment by Chuxel** [comment, marked as answer]
In short, yes. VS Code workspace files provide a number of advantages including settings, support for triggering .vscode/launch.json or .vscode/tasks.json from across each root, and the ability to more spawn terminals under each sub-root,

We do not have a way to automatically jump into one currently, but you'll be notified to open it if a `.code-workspace` file is in the root of the repository.

If you want extensions to always be installed, you can use the `extensions` property in `.devcontainer/devcontainer.json`.  Extension recommendations are there when you do not want to force people to install something and leave the installation up to personal preference.

**Reply by brwilkinson** [reply]
> We do not have a way to automatically jump into one currently, but you'll be notified to open it if a .code-workspace file is in the root of the repository.

Yes this works, I was hoping or thinking of a way to automatically open it.

Thanks for the reply and info. That answers my question.

**Reply by gabynevada** [reply]
@Chuxel currently even if we have the .code-workspace in the root of the repo we don't get any notification from VSCode

**Reply by dijonkitchen** [reply]
I also recently made a codespace with a `.code-workspace` file, but I see no notification and it doesn't open automatically.

**Reply by chamaoskurumi** [reply]
me neither

**Reply by dijonkitchen** [reply]
Noting here that the only way I've seen it work is to manually go to the top menu: File > Open Workspace from File... and select your `.code-workspace` file. 

<img width="278" height="195" alt="image" src="https://github.com/user-attachments/assets/677f64ff-9999-4027-8a4d-699c319bb9af" />
